embassy-macros/src/lib.rs

```diff
@@ -250,7 +250,11 @@ pub fn interrupt_take(item: TokenStream) -> TokenStream {
             static TAKEN: ::embassy::export::atomic::AtomicBool = ::embassy::export::atomic::AtomicBool::new(false);
 
             if TAKEN.compare_exchange(false, true, ::embassy::export::atomic::Ordering::AcqRel, ::embassy::export::atomic::Ordering::Acquire).is_err() {
-                panic!("IRQ Already taken");
+                #[cfg(feature = "defmt")]
+                defmt::panic!("IRQ Already taken");
```

Comment:
the `cfg`s inside `quote!()` get included in the macro output literally, so they're checking for the `defmt` feature in the user's crate, not embassy-macros.  You should check them outside.

One problem is it'll emit code using `defmt::panic`, so `defmt` has to be in scope in the user's crate. This will break if the user enables `defmt` in embassy but doesn't depend on `defmt` themself. This should be rare though, so maybe it's still worth it...?

Reply:
Closing now, this was resolved and fixed according to the discussion on Matrix - always use `core::panic`.